testsuite/integration-arquillian/servers/auth-server/services/testsuite-providers/src/main/java/org/keycloak/testsuite/federation/PassThroughFederatedUserStorageProvider.java

```diff
@@ -91,18 +96,19 @@ public boolean isValid(RealmModel realm, UserModel user, CredentialInput input)
     public boolean updateCredential(RealmModel realm, UserModel user, CredentialInput input) {
         // testing federated credential attributes
         if (input.getType().equals(PasswordCredentialModel.TYPE)) {
-            List<CredentialModel> existing = session.userFederatedStorage().getStoredCredentialsByType(realm, user.getId(), "CLEAR_TEXT_PASSWORD");
-            if (existing.isEmpty()) {
```

Comment:
You switched the content between `if` and `else` blocks on purpose?

Reply:
Yeah, do you think I should have preserved the original logic instead?

Reply:
No it's OK. I didn't notice the difference in `existing.isEmpty()` vs `existing.isPresent()` :)